pkg/gds/emails/emails_test.go

```diff
@@ -111,23 +112,42 @@ func TestSendEmails(t *testing.T) {
 	receipient, err := mail.ParseAddress(conf.AdminEmail)
 	require.NoError(t, err)
 
+	var expected []byte
 	vcdata := emails.VerifyContactData{Name: receipient.Name, Token: "Hk79ZIhCSrYJtSaaMECZZKI1BtsCY9zDLPq9c1amyK2zJY6T", VID: "9e069e01-8515-4d57-b9a5-e249f7ab4fca", BaseURL: "http://localhost:3000/verify-contact"}
 	msg, err := emails.VerifyContactEmail(sender.Name, sender.Address, receipient.Name, receipient.Address, vcdata)
 	require.NoError(t, err)
 	require.NoError(t, email.Send(msg))
+	require.Len(t, emails.MockEmails, 1)
+	expected, err = json.Marshal(msg)
+	require.NoError(t, err)
+	require.Equal(t, expected, emails.MockEmails[0])
 
 	rrdata := emails.ReviewRequestData{Request: "foo", Token: "abcdef1234567890", VID: "42", Attachment: []byte(`{"hello": "world"}`)}
 	msg, err = emails.ReviewRequestEmail(sender.Name, sender.Address, receipient.Name, receipient.Address, rrdata)
 	require.NoError(t, err)
 	require.NoError(t, email.Send(msg))
+	require.Len(t, emails.MockEmails, 2)
+	expected, err = json.Marshal(msg)
+	require.NoError(t, err)
+	require.Equal(t, expected, emails.MockEmails[1])
 
 	rjdata := emails.RejectRegistrationData{Name: receipient.Name, Reason: "not a good time", VID: "42"}
 	msg, err = emails.RejectRegistrationEmail(sender.Name, sender.Address, receipient.Name, receipient.Address, rjdata)
 	require.NoError(t, err)
 	require.NoError(t, email.Send(msg))
+	require.Len(t, emails.MockEmails, 3)
+	expected, err = json.Marshal(msg)
+	require.NoError(t, err)
+	require.Equal(t, expected, emails.MockEmails[2])
 
 	dcdata := emails.DeliverCertsData{Name: receipient.Name, VID: "42", CommonName: "example.com", SerialNumber: "1234abcdef56789", Endpoint: "trisa.example.com:443"}
 	msg, err = emails.DeliverCertsEmail(sender.Name, sender.Address, receipient.Name, receipient.Address, "testdata/foo.zip", dcdata)
 	require.NoError(t, err)
 	require.NoError(t, email.Send(msg))
+	require.Len(t, emails.MockEmails, 4)
+	expected, err = json.Marshal(msg)
+	require.NoError(t, err)
+	require.Equal(t, expected, emails.MockEmails[3])
+
+	t.Cleanup(emails.PurgeMockEmails)
```

Comment:
Nice use of the cleanup function! I believe this should go at the top of the test function to guarantee it gets run though. I'm not 100% sure, but I suspect that the `require` functions execute a `t.FailNow()` which stops the execution of the test, so if any of the tests fail, the cleanup function won't be run since it's the last function call.